app/scripts/metamask-controller.js

```diff
@@ -6517,76 +6255,50 @@ export default class MetamaskController extends EventEmitter {
       getNonEvmSupportedMethods: this.getNonEvmSupportedMethods.bind(this),
     });
 
-    // Handle Solana account notifications
     // The optional chain operator below shouldn't be needed as
     // the existence of sessionProperties is enforced by the caveat
     // validator, but we are still seeing some instances where it
     // isn't defined in production:
     // https://github.com/MetaMask/metamask-extension/issues/33412
     // This suggests state corruption, but we can't find definitive proof that.
     // For now we are using this patch which is harmless and silences the error in Sentry.
-    const solanaAccountsChangedNotifications =
-      caip25Caveat.value.sessionProperties?.[
-        KnownSessionProperties.SolanaAccountChangedNotifications
-      ];
-    const solanaScope =
-      sessionScopes[MultichainNetworks.SOLANA] ||
-      sessionScopes[MultichainNetworks.SOLANA_DEVNET] ||
-      sessionScopes[MultichainNetworks.SOLANA_TESTNET];
-
-    if (solanaAccountsChangedNotifications && solanaScope) {
-      const { accounts } = solanaScope;
-      const parsedPermittedSolanaAddresses = accounts.map((caipAccountId) => {
-        const { address } = parseCaipAccountId(caipAccountId);
-        return address;
-      });
-
-      const [accountAddressToEmit] = this.sortMultichainAccountsByLastSelected(
-        parsedPermittedSolanaAddresses,
-      );
+    NON_EVM_ACCOUNT_CHANGED_CONFIGS.forEach(
+      ({ network, chains, notificationProperty }) => {
+        const accountsChangedNotifications =
+          caip25Caveat.value.sessionProperties?.[notificationProperty];
 
-      if (accountAddressToEmit) {
-        this._notifyMultichainAccountChange(
-          origin,
-          [accountAddressToEmit],
-          MultichainNetworks.SOLANA,
-        );
-      }
-    }
+        if (!accountsChangedNotifications) {
+          return;
+        }
 
-    ///: BEGIN:ONLY_INCLUDE_IF(tron)
-    // Handle Tron account notifications
-    const tronAccountsChangedNotifications =
-      caip25Caveat.value.sessionProperties?.[
-        KnownSessionProperties.TronAccountChangedNotifications
-      ];
+        // Find the scope by checking all chains in the config
+        const scopeObject = chains
+          .map((chain) => sessionScopes[chain])
+          .find((scope) => scope !== undefined);
```

Comment:
this will only check the accounts from one scopeObject that matches the `chains`. I guess that doesn't really matter with how we're currently setting the same accounts on all the scopeObjects, but i'm worried if we ever change that that we would introduce a bug here